### ISSUE TITLE
Should results merging happen in CI?

### ISSUE THREAD

**Issue by gsnedders** [web-platform-tests/wpt.fyi#3980]
Currently, the results processor receives either: a list of wptreport JSON URLs and a list of screenshot URLs, or a single ZIP URL, where the ZIP contains a number of wptreport JSON files and a number of screenshot files.

The results processor then attempts to merge all these files together.

We currently have a bunch of things we ignore when merging all the files together:

https://github.com/web-platform-tests/wpt.fyi/blob/5682248b058b181e1a222595a8125168ff7a3f55/results-processor/wptreport.py#L38-L41

Ideally we'd put the logic somewhere closer to CI where we can actually ignore fields that make sense for that specific configuration.

GitHub Actions also _always_ compresses artifacts, and if we're using multiple jobs for parallelism (the only sensible approach!), then we end up with multiple ZIP files which need to get merged (which we don't currently support).

**Comment by past** [comment]
I'm OK with this in principle, but it means that we have to make this change across all CIs, right?

**Comment by gsnedders** [comment]
@past Right. I'd imagine we create some `wpt merge-results` or something where the logic actually lives, but obviously this involves more intermediate artifacts being created in every CI system.

cc @jgraham

**Comment by jgraham** [comment]
I'm basically happy with the idea of CI being taught to merge the results of multiple runs into one giant JSON file; having all the detailed results from a run in one place can sometimes be useful independent of wpt.fyi, so I think that is enough to make this worth doing.
